File: VB/XtraReport1.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Data
Imports System.Windows.Forms
Imports DevExpress.XtraReports.UI
' ...

Namespace GetDataInPreview
	Partial Public Class XtraReport1
		Inherits DevExpress.XtraReports.UI.XtraReport
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub xrLabel1_BeforePrint(ByVal sender As Object, ByVal e As System.Drawing.Printing.PrintEventArgs) Handles xrLabel1.BeforePrint
			CType(sender, XRLabel).Tag = GetCurrentRow()
		End Sub

		Private Sub xrLabel1_PreviewClick(ByVal sender As Object, ByVal e As PreviewMouseEventArgs) Handles xrLabel1.PreviewClick
			Dim dataRow As DataRowView = TryCast(e.Brick.Value, DataRowView)
			MessageBox.Show(dataRow.Row("ShipCountry").ToString() & ": " & e.Brick.Text)
		End Sub

	End Class
End Namespace

File: CS/XtraReport1.cs
using System;
using System.Data;
using System.Windows.Forms;
using DevExpress.XtraReports.UI;
// ...

namespace GetDataInPreview {
    public partial class XtraReport1 : DevExpress.XtraReports.UI.XtraReport {
        public XtraReport1() {
            InitializeComponent();
        }

        private void xrLabel1_BeforePrint(object sender, System.Drawing.Printing.PrintEventArgs e) {
            ((XRLabel)sender).Tag = GetCurrentRow();
        }

        private void xrLabel1_PreviewClick(object sender, PreviewMouseEventArgs e) {
            DataRowView dataRow = e.Brick.Value as DataRowView;
            MessageBox.Show(dataRow.Row["ShipCountry"].ToString() + ": " + e.Brick.Text);
        }

    }
}
